internal/controller/controllers/hypershiftagentserviceconfig_controller_test.go

```diff
@@ -155,4 +183,30 @@ var _ = Describe("HypershiftAgentServiceConfig reconcile", func() {
 			Expect(err.Error()).To(ContainSubstring("Failed to create client"))
 		})
 	})
+
+	Context("Ensure agent-install CRDs exist on spoke clister", func() {
```

Comment:
You don't need a context here as you're not adding any shared context.

Also you should add a test for update.

Reply:
Maybe 'Describe' instead? Just for grouping these separately.